### DISCUSSION TITLE
set column width

### DISCUSSION THREAD

**Discussion by akehat** [rappasoft/laravel-livewire-tables#402] [Q&A]
How do I set a column width so that it's not dynamic width, but always X%?

**Comment by rappasoft** [comment]
Can try: https://github.com/rappasoft/laravel-livewire-tables/wiki/Misc.-Column-Functionality#adding-any-attribute-to-the-column-headers

**Comment by jadersbr** [comment]
Hi, currently Is there a way to set the width of a column in %?

**Comment by lrljoe** [comment]
The below would set width of w-8 (tailwind style) for the "name" column.  You'll need to add  any  other required styling from the default set to ensure that it doesn't look too odd, including your "whitespace-normal" etc
```
       $this->setTdAttributes(function (Column $column, $row, $columnIndex, $rowIndex) {
             if ($column->isField('name') && $row->total < 500) {
                return [
                    'default' => false,
                    'class' => 'w-8',
                ];
            }
            return [];
        });
```

You can of course create a class for a specific width % and use that instead.
Using style inline is not best practice.
